core/src/main/java/greencity/converters/UserArgumentResolver.java

```diff
@@ -34,14 +31,13 @@ public boolean supportsParameter(MethodParameter parameter) {
 
     /**
      * Method returns {@link User} by principal.
+     *
      * @return {@link User}
      */
     @Override
     public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer mavContainer,
                                   NativeWebRequest webRequest, WebDataBinderFactory binderFactory) {
         Principal principal = webRequest.getUserPrincipal();
-        return Optional.ofNullable(userService.findByEmail(Objects.requireNonNull(principal).getName()))
-            .orElseThrow(() -> new WrongEmailException(ErrorMessage.USER_NOT_FOUND_BY_EMAIL + principal.getName())
-            );
+        return userService.findByEmail(principal != null ? principal.getName() : null);
```

Comment:
Please do not search by `null`. There is no reason to call `findByEmail` with argument equal to `null` if you know the result in advance.